src/client/containers/WelcomePage/WelcomePage.container.js

```diff
@@ -1,5 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
+import './WelcomePage.styles.css';
+import GenericButton from '../../components/GenericButton/GenericButton.component';
+import GoogleButton from '../../components/GoogleButton/GoogleButton.component';
+import { useFirebase } from '../../firebase/FirebaseContext';
+import { Redirect } from 'react-router-dom';
 
 export default function Welcome() {
-  return <div>This is your private Welcome</div>;
+  const [redirect, setRedirect] = useState();
+  const { signIn, auth } = useFirebase();
+
+  function addUser() {
+    (async () => {
+      const token = await auth.currentUser.getIdToken();
+      await fetch('/api/users', {
+        method: 'POST',
+        mode: 'cors',
+        headers: {
+          'Content-type': 'application/json',
+          authorization: `Bearer ${token}`,
+        },
+      });
+    })();
+  }
+
+  const signInUser = async () => {
+    await signIn();
+    addUser();
```

Comment:
Here the post request will have to return the user information. The id must then be stored in localstorage for everyone to use

Reply:
Sorry, what we need here is to use a context. Same as the firebase context and then wrap around the app

**UserProvider.js**
```
import PropTypes from 'prop-types';
import React, { createContext, useContext, useEffect } from 'react';
import { useAuthentication } from '../hooks/useAuthentication';

const UserContext = createContext();

export function UserProvider({ children }) {
  const { isAuthenticated } = useAuthentication();

  // post request to create user when isAuthenticated is true
  // store user in local storage
  // set the provider value to user

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
}

export function useUser() {
  const user = useContext(UserContext);

  if (!user) {
    throw new Error('This component must be under UserProvider');
  }

  return user;
}

```

Then the AppContainer looks like this:

```
  <ErrorBoundary>
      <FirebaseProvider>
        <UserProvider>
          <App />
        </UserProvider>
      </FirebaseProvider>
    </ErrorBoundary>
```

Reply:
...and I think thats an issue on its own.

Reply:
So should we do another PR for that?

Reply:
Yes, lets do that. Then its separate gitwise and codewise. You will have to remove the post request in this pr